workflows/dags/literature/hep_create_dag.py

```diff
@@ -342,6 +344,115 @@ def normalize_journal_titles(**context):
                 overwrite=True,
             )
 
+        @task
+        def refextract(**context):
+            """Extract references from various sources and add them to the workflow.
+
+            Runs ``refextract`` on both the PDF attached to the workflow and the
+            references provided by the submitter, if any, then chooses the one
+            that generated the most and attaches them to the workflow object.
+
+            Returns:
+                None
+            """
+
+            workflow_id = context["params"]["workflow_id"]
+
+            workflow_data = read_object(s3_hook, bucket_name, workflow_id)
+
+            source = LiteratureReader(workflow_data["data"]).source
+
+            response = inspire_http_hook.call_api(
+                endpoint="api/matcher/journal-kb",
+                method="GET",
+            )
+            response.raise_for_status()
+            journal_kb_dict = get_value(response.json(), "journal_kb_data")
+
+            if "references" in workflow_data["data"]:
+                raw_refs_to_extract, references = raw_refs_to_list(
+                    workflow_data["data"]["references"]
+                )
+                extracted_references_dict = extract_references_from_list(
+                    raw_refs_to_extract["values"], journal_kb_data=journal_kb_dict
+                )
+                extracted_references = dedupe_list(
+                    extracted_references_dict["extracted_references"]
+                )
+                logger.info(
+                    "Extracted %d references from raw refs.", len(extracted_references)
+                )
+
+                response = inspire_http_hook.call_api(
+                    endpoint="api/matcher/linked_references/",
+                    method="POST",
+                    json={"references": extracted_references + references},
+                )
+                response.raise_for_status()
+
+                workflow_data["data"]["references"] = response.json().get(
+                    "references", []
+                )
+
+                write_object(
+                    s3_hook,
+                    workflow_data,
+                    bucket_name,
+                    key=workflow_id,
+                    overwrite=True,
+                )
+
+                return
+
+            matched_pdf_references = []
+
+            key = workflows.get_document_key_in_workflow(workflow_data)
+            if not key:
+                logger.info("No document found for reference extraction.")
+                return
+
+            with TemporaryDirectory(prefix="refextract") as tmp_dir:
```

Comment:
we are missing the actual case -> https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/tasks/actions.py#L524-L529. The code here represents the else scenario https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/tasks/actions.py#L530-L534, which  i am not even sure if it is ever called. This part will be called only if there is not url https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/tasks/actions.py#L522 . Is that ever the case ?